src/theoretical.jl

```diff
@@ -64,6 +64,27 @@ Evaluate the variogram at points `x` and `y`.
 #------------------
 # IMPLEMENTATIONS
 #------------------
+"""
+    NuggetEfect(n)
+
+A pure nugget effect (random) variogram with nugget `n`
+"""
+@with_kw struct NuggetEffect{T,D} <: Variogram{T,D}
+  nugget::T = 0.0
+  sill::T = 0.0
+  range::T = 0.0
+  distance::D = Euclidean()
+end
+function NuggetEffect(n::T) where {T <: AbstractFloat}
+  @assert n > 0 "Nugget effect should be greater than zero"
+  NuggetEffect(nugget=n)
+end
+
+(γ::NuggetEffect{T,D})(h) where {T,D} = (h > 0) ? γ.nugget : zero(T)
```

Comment:
Please write this as `(h > 0) * γ.nugget` as we discussed in the issue. You can then eliminate the type parameters T and D like with the other models.

Reply:
OK, they are almost equivalent. Although the proposal saves one multiplication :-)

Reply:
It also makes the code less readable. Let's keep things simple, and follow the conventions in the code.